File: AdventOfCode2022/Day1.cs
namespace AdventOfCode2022 {
    internal static class Day1 {
        public static void Run(string path) {
            Console.WriteLine("Day One");
            string input = FileHelper.ReadFileAsString(path);

            Solve(input);
        } 

        private static void Solve(string input) {
            string[] splitted = input.Split("\n\n");
            List<int> calories = new();

            foreach (var item in splitted)
            {
                var calorie = item
                    .Split("\n", StringSplitOptions.RemoveEmptyEntries)
                    .Select(x => int.Parse(x))
                    .Sum();

                calories.Add(calorie);  
            }

            Console.WriteLine($"Max calorie: {calories.Max()}");
            Console.WriteLine($"Top three: {calories.OrderByDescending(x => x).Take(3).Sum()}");
        }
    }
}

File: AdventOfCodeFsharp/Day1.fs
module Day1

open System.IO
open System

let Solve(path: string) =
    printfn "Day One"

    let calories = 
        File.ReadAllText(path).Split("\n\n") 
        |> Seq.map(fun x -> x.Split("\n", StringSplitOptions.RemoveEmptyEntries) |> Seq.map(int)) 
        |> Seq.map(Seq.sum)
    
    let max = calories |> Seq.max
    let top = calories |> Seq.sortDescending |> Seq.take(3) |> Seq.sum

    printfn "Max calorie: %i" max
    printfn "Top calories: %i" top